index.rst

```diff
@@ -44,8 +44,515 @@
+        description
+        api_version
+        auth
+    }
+
+The fields ``name``, ``description``, and ``version`` are arbitrary.
+`semantic_versioning`_ is strongly encouraged for ``version``.
+``api_version`` should reflect the version of the API in use (at time of
+writing, ``1.0``).  ``repository`` is the URL for the source of your
+project.  If you want your microservice to be published on
+``api.lsst.codes`` its source must be publicly available.  We extremely
+strongly recommend hosting it on GitHub.
+
+The ``auth`` field is constrained.  It must be one of ``none``,
+``basic``, or ``bitly-proxy``.  These represent the three choices
+available to the microservice for authentication to GitHub (we have
+standardized on GitHub as a canonical source of authentication data,
+since LSST is fairly fundamentally coupled to GitHub and it functions as
+a widely available OAuth2 provider):
+
+ - ``none``: no authentication required.
+ - ``basic``: HTTP Basic Auth.  Typically used with a GitHub username and
+   token, although if you didn't have two-factor authentication enabled
+   at GitHub you could use a password here as well.
+ - ``bitly-proxy``: Authenticate through the Bitly OAuth2 proxy.
+   Typically used with a GitHub username and password, and basically
+   converts two-factor authentication back into username-and-password
+   authentication.
+
+The good news is, if you're writing in Python and your application is a
+`Flask <http://flask.pocoo.org/>`_  app, you don't need to implement the
+metadata route.  Just use ``apikit``. 
+
+Using apikit
+------------
+
+The ``apikit`` module is documented at `github_apikit`_.  ``apikit`` has
+two classes: :py:class:`apikit.APIFlask` and
+:py:class:`apikit.BackendError`, and two functions:
+``set_flask_metadata()`` and ``add_metadata_route()``.
+
+The :py:class:`apikit.APIFlask` class is what you should generally use: it
+is a subclass of a Flask application (:py:class:`flask.Flask`) that
+already has metadata added and the route baked into it.
+
+If you have an existing Flask application, you might want to use
+``apikit.set_flask_metadata()`` on that application rather than the
+:py:class:`apikit.APIFlask` class.  You will find ``add_route_prefix()``
+useful to add additional routes to the metadata.  That is helpful, for
+instance, for Kubernetes Ingress resources, which provide routing but
+not path rewriting, which makes it your responsibility to ensure the
+metadata is available at ``/{{app_name}}/metadata`` as well as
+``/metadata``.
+
+The :py:class:`apikit.BackendError` class is useful with Flask decorators
+to return diagnostic information when something goes wrong with your
+application.  You'll see it in the example below.
+
+Example apikit usage
+^^^^^^^^^^^^^^^^^^^^
+
+The following describes how you would use ``apikit`` and specifically
+:py:class:`apikit.APIFlask` to create a service wrapper suitable for use
+on ``api.lsst.codes``.
+
+Microservice server
+@@@@@@@@@@@@@@@@@@@
```

Comment:
`@` -> `"` — https://developer.lsst.io/docs/rst_styleguide.html#sections

I wonder if you can add more sections at the level of "microservice server" but for different aspects of the server, rather than having a monolithic section.